File: matlab/lib/+Ice/TCPEndpointInfo.m
classdef (Sealed) TCPEndpointInfo < Ice.IPEndpointInfo
    %TCPENDPOINTINFO Provides access to a TCP endpoint information.
    %
    %   TCPEndpointInfo Methods:
    %     type - Returns the type of the endpoint.
    %     secure - Returns true if this endpoint is a secure endpoint.

    % Copyright (c) ZeroC, Inc.

    methods(Hidden)
        function obj = TCPEndpointInfo(compress, host, port, sourceAddress, type, secure)
            assert(nargin == 6, 'Invalid number of arguments');
            obj@Ice.IPEndpointInfo(compress, host, port, sourceAddress);
            obj.type_ = type;
            obj.secure_ = secure;
        end
    end
    methods
        function r = type(obj)
            r = obj.type_;
        end

        function r = secure(obj)
            r = obj.secure_;
        end
    end
    properties(GetAccess=private, SetAccess=immutable)
        type_ (1, 1) int16
        secure_ (1, 1) logical
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/TCPEndpointInfo.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/**
 * Provides access to a TCP endpoint information.
 *
 * @see Endpoint
 */
public final class TCPEndpointInfo extends IPEndpointInfo {
    private final short _type;
    private final boolean _secure;

    @Override
    public short type() {
        return _type;
    }

    @Override
    public boolean secure() {
        return _secure;
    }

    // internal constructor
    TCPEndpointInfo(
            int timeout,
            boolean compress,
            String host,
            int port,
            String sourceAddress,
            short type,
            boolean secure) {
        super(timeout, compress, host, port, sourceAddress);
        this._type = type;
        this._secure = secure;
    }
}
